### PR TITLE
NUNM simplified

### PR DESCRIPTION
**Pull request by lvmcerisy**
version of NUNM with simplifications, better performance, less code rewriting
passes the tests and is consistent with specific sterile 4 scenario as a crosscheck

### CODE REVIEW

File: test/Utils.h [hunk 1]
```diff
@@ -143,12 +143,12 @@ OscProb::PMNS_NUNM* GetNUNM(bool is_nominal){
...
-    p->SetAlpha(2,0, 0.3, 0);
-    p->SetAlpha(1,1, 0.4, 0);
-    p->SetAlpha(2,1, 0.5, 0);
-    p->SetAlpha(2,2, 0.6, 0);
+    p->SetAlpha(0,0, 0.05, 0);
+    p->SetAlpha(1,0, 0.06, 0);
+    p->SetAlpha(2,0, 0.07, 0);
+    p->SetAlpha(1,1, 0.08, 0);
+    p->SetAlpha(2,1, 0.09, 0);
+    p->SetAlpha(2,2, 0.1, 0);
```
Comment (joaoabcoelho) [inline test/Utils.h:151]: Why did you change the test values here? It's not a big problem, but if you kept the same values you could have kept the same ROOT test file to compare that probabilities are identical. Did you check that the values were indeed the same before you changed this?

Comment (lvmcerisy) [reply test/Utils.h:151]: I have changed them to match the values used to test the sterile model. I do not expect them to be identical to before because the new implementation changes the way the probabilities is propagated, this recovers consistency with sterile when also the off diag pars are non zero.

File: inc/PMNS_NUNM.h [hunk 2]
```diff
@@ -56,20 +56,22 @@ namespace OscProb {
...
       virtual void PropagatePath(NuPath p);
       double       fracVnc; // set fraction of matter potential affecting NC
+      vectorC      ApplyAlphaDagger(vectorC fState);
+      vectorC      ApplyAlpha(vectorC fState);
       void         InitMatrix();
+      vectorC fNuStateBuffer;
       Eigen::Matrix<std::complex<double>, 3, 3> X;
       Eigen::Matrix<std::complex<double>, 3, 3> Alpha;
       Eigen::Matrix<std::complex<double>, 3, 3> V;
       Eigen::Matrix<std::complex<double>, 3, 3> Ham;
```
[the remark this answers is not in the document]
Comment (lvmcerisy) [reply inc/PMNS_NUNM.h]: Incorporated now.

File: test/Utils.h (hunk 1 above)
Comment (joaoabcoelho) [reply test/Utils.h:151]: Alright, so this means we have a bug fix in this pull request. Duly noted. And indeed we shouldn't expect consistency with the previous version.
